pkg/asset/cluster/alibabacloud/alibabacloud.go

```diff
@@ -10,5 +10,6 @@ func Metadata(config *types.InstallConfig) *alibabacloud.Metadata {
 	return &alibabacloud.Metadata{
 		Region:        config.Platform.AlibabaCloud.Region,
 		ClusterDomain: config.ClusterDomain(),
+		PrivateZoneID: config.AlibabaCloud.PrivateZoneID,
```

Comment:
Generally we try to avoid adding data to the Metadata when it is not necessary, but I do not see another way of determining for certain that the user provided the private dns zone.